### PR TITLE
new order jobs following repository-list.txt

### PR DESCRIPTION
**Pull request by mbiarnes**
**Thank you for submitting this pull request**

- changed the dependant projects following the https://github.com/kiegroup/droolsjbpm-build-bootstrap/blob/main/docs/project-dependencies-hierarchy.png
- disabled optaplanner main PR since this is done in kogito
- the opta* - 7.x deploy jobs are also in the same folder as the other for main

<pre>
How to retest a PR or trigger a specific build:

* <b>a pull request</b> please add comment: <b>Jenkins retest this</b>
</pre>

### CODE REVIEW

File: job-dsls/jobs/kie/main/deploy_jobs.groovy [hunk 1]
```diff
@@ -54,56 +62,35 @@ def final REPO_CONFIGS = [
...
-                        "full"                     : "true",
-                        "integration-tests"        : "true",
-                        "maven.test.failure.ignore": "true"
-                ]
+                downstreamRepos        : ["kie-wb-common"]
         ],
         "jbpm"                      : [
                 timeoutMins            : 120,
                 mvnGoals               : DEFAULTS["mvnGoals"] + " -Dcontainer.profile=wildfly",
-                downstreamRepos        : ["jbpm-work-items", "kie-jpmml-integration"]
```
Comment (mareknovotny) [inline job-dsls/jobs/kie/main/deploy_jobs.groovy:78]: I think this is not right to remove.

File: job-dsls/jobs/kie/main/deploy_jobs.groovy [hunk 2]
```diff
@@ -54,56 +62,35 @@ def final REPO_CONFIGS = [
...
+                downstreamRepos        : ["kie-jpmml-integration"]
         ],
         "kie-jpmml-integration"     :[
                 downstreamRepos        : ["droolsjbpm-integration"]
```
Comment (mareknovotny) [inline job-dsls/jobs/kie/main/deploy_jobs.groovy:77]: This is also wrong, there is no downstream repo for kie-jpmml-integration. And droolsjbpm-integration should be in jbpm downstreamRepos too

**Review by mareknovotny — CHANGES_REQUESTED** [review]
there are some weird deps in downstreamRepos like for jbpm and for kie-jpmml-integration

**Comment by mbiarnes** [comment]
@mareknovotny I tried to follow this image: https://github.com/kiegroup/droolsjbpm-build-bootstrap#kiegroup-project-structure
there it says that `kie-jpmml-integration` is following `jbpm` and that `jbpm-workitems` comes after `droolsjbpm-integration`

File: job-dsls/jobs/kie/main/deploy_jobs.groovy [hunk 3]
```diff
@@ -288,6 +281,11 @@ for (repoConfig in REPO_CONFIGS) {
                 }
             }
 
+            def downstreamRepos = get("downstreamRepos")
+            if (downstreamRepos) {
+                downstream(downstreamRepos, 'UNSTABLE')
```
Comment (Ginxo) [inline job-dsls/jobs/kie/main/deploy_jobs.groovy:286]: @mbiarnes I don't know where this `downstrea` method is defined. Can you please clarify?

**Comment by mareknovotny** [comment]
> @mareknovotny I tried to follow this image: https://github.com/kiegroup/droolsjbpm-build-bootstrap#kiegroup-project-structure
> there it says that `kie-jpmml-integration` is following `jbpm` and that `jbpm-workitems` comes after `droolsjbpm-integration`

I don't think https://github.com/kiegroup/droolsjbpm-build-bootstrap/blob/main/.ci/project-dependencies.yaml#L61 is correct, but that needs to be checked or verified with a build? @Ginxo you added that dependency, but what project requested it?

File: job-dsls/jobs/kie/main/deploy_jobs.groovy (hunk 3 above)
Comment (mareknovotny) [reply job-dsls/jobs/kie/main/deploy_jobs.groovy:286]: that is DSL plugin method @Ginxo see our jenkins APi <jenkins_URL>/plugin/job-dsl/api-viewer/index.html#method/javaposse.jobdsl.dsl.helpers.publisher.PublisherContext.downstream

Comment (mbiarnes) [reply job-dsls/jobs/kie/main/deploy_jobs.groovy:286]: @Ginxo this downstream repos definition does:
image
Please take a look at 
https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/custom/job/mbiarnes/job/KIE/job/main/job/deployedRep/
All jobs are disabled! But this is this PR.

**Comment by mareknovotny** [comment]
the audit for dependencies is needed as a followup in https://issues.redhat.com/browse/BXMSPROD-1444

**Review by mareknovotny — APPROVED** [review]

**Pull request merged**